cmd/system-probe/modules/oom_kill_probe.go

```diff
@@ -23,18 +24,20 @@ var OOMKillProbe = module.Factory{
 		if err != nil {
 			return nil, fmt.Errorf("unable to start the OOM kill probe: %w", err)
 		}
-		return &oomKillModule{okp}, nil
+		return &oomKillModule{OOMKillProbe: okp}, nil
 	},
 }
 
 var _ module.Module = &oomKillModule{}
 
 type oomKillModule struct {
 	*probe.OOMKillProbe
+	lastCheck time.Time
 }
 
 func (o *oomKillModule) Register(httpMux *module.Router) error {
 	httpMux.HandleFunc("/check/oom_kill", func(w http.ResponseWriter, req *http.Request) {
+		o.lastCheck = time.Now()
```

Comment:
I believe you need to protect this with a mutex

Reply:
This is only used for status reporting, so I didn't think it was important enough to protect with a mutex. Do you agree/disagree?

Reply:
I disagree because I think this could be more dangerous than just stale data for status reporting. I know that in the event of concurrent read/writes for _maps_ the Go runtime will literally panic and crash, so I would avoid have any sort of known data race.

Reply:
This isn't a map, and `GetStats` returns a new map each time, so there is no concurrent read/writes on a map.

Reply:
Regardless, I will fix this. I think a mutex might be heavy-handed, so I'm going to use atomics.

Reply:
What I'm trying to say is that this _is_ an instance of data race (you're reading and writing to this field from different go-routines), and that Go can even panic in certain scenarios. True, I'm not sure if the same behavior given to maps scenario applies to this, but in any case why would risk that and ship a bug? You can either add a mutex or just use an atomic to store the timestamp.